### PR TITLE
improve message abstraction to reflect semantics after message is parsed

### LINKED ISSUE #556: improve message abstraction to reflect semantics after message is parsed
I found it easier to describe the different message sections if we talk about the abstract data structures produced as a result of parsing, as opposed to the specific parsing mechanisms of 1.1.

### PR DESCRIPTION
**Pull request by royfielding**
Fixes #556

### CODE REVIEW

**Comment by royfielding** [comment]
This is also related to #294

**Review by reschke — APPROVED** [review]

File: draft-ietf-httpbis-semantics-latest.xml [hunk 1]
```diff
@@ -2105,22 +2158,24 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
...
 </t>
 <t>
-   <iref item="header section"/>
-   Fields that are sent/received before the message body are referred to as
-   "header fields" (or just "headers", colloquially) and are located within the
-   "<x:dfn>header section</x:dfn>" of a message. We refer to some named fields
-   specifically as a "header field" when they are only allowed to be sent in
-   the header section.
+   The "<x:dfn>header section</x:dfn>" of a message consists of a sequence of
+   of header fields. Each header field might modify or extend message
```
Comment (mnot) [inline draft-ietf-httpbis-semantics-latest.xml:2169]:
```suggestion
   of header field lines. Each header field might modify or extend message
```

File: draft-ietf-httpbis-semantics-latest.xml [hunk 2]
```diff
@@ -2014,24 +2023,48 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
...
+   obtained while sending the payload.
+</t>
 <t>
-   Each major version of HTTP defines its own syntax for the communication of
-   messages. However, they share a common data abstraction.
+   Framing and control data is sent first, followed by a header section
+   containing fields for the headers table. When a message includes a payload,
+   the payload data is sent after the header section and potentially
+   interleaved with zero or more trailer sections containing fields for the
+   trailers table.
```
Comment (mnot) [inline draft-ietf-httpbis-semantics-latest.xml:2050]: I think we should explicitly note that trailers are constrained in some protocol versions.

**Review by mnot — CHANGES_REQUESTED** [review]
Left a couple of notes, otherwise looks good.

File: draft-ietf-httpbis-semantics-latest.xml (hunk 2 above)
Comment (royfielding) [reply draft-ietf-httpbis-semantics-latest.xml:2050]: We do that in the section on Trailers, to the extent they are constrained. HTTP/2 and 3 do not explicitly constrain trailers, they just don't have a frame to carry them yet.

Comment (royfielding) [reply draft-ietf-httpbis-semantics-latest.xml:2050]: https://github.com/httpwg/http-core/pull/557/commits/3a96888c38b67b28e045f75fc111ac38fb7e5fb7

**Review by mnot — APPROVED** [review]

**Review by reschke — APPROVED** [review]

**Pull request merged**